rca/wagtailapi/filters.py

```diff
@@ -19,6 +22,43 @@ def filter_queryset(self, request, queryset, view):
         return queryset
 
 
+class SubjectsFilter(filters.BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        if hasattr(queryset.model, "subjects"):
+            subject_ids = request.GET.getlist("subjects", [])
+            if subject_ids:
+                queryset = queryset.model.objects.filter(
+                    subjects__subject__in=subject_ids
+                )
+
+        return queryset
+
+
+class RelatedSchoolsFilter(filters.BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        if hasattr(queryset.model, "related_schools_and_research_pages"):
+            school_ids = request.GET.getlist("related_schools_and_research_pages", [])
```

Comment:
Sanitise the input, e.g. `[int(id) for id in request.GET.getlist("related_schools_and_research_pages", [])]`